kong/db/dao/init.lua

```diff
@@ -1494,6 +1509,9 @@ function DAO:cache_key(key, arg2, arg3, arg4, arg5, ws_id)
   -- the generic code below, but building the cache key
   -- becomes a single string.format operation
   if type(key) == "string" then
+
+    key = normalize_string(key)
```

Comment:
Should we do this inside the LMDB specific codepath instead? Doing this for non-LMDB codepath seems to make little sense.

Reply:
use `self.db.strategy == off` to determine if  LMDB specific codepath

Reply:
Wouldn't keeping the behavior consistent across all code paths be better here?
Datong, is there a specific concern here?

Reply:
I also agree with @hbagdi , we should have the same code path.

Reply:
@dndx, could you confirm this change? I think the condition `strategy == "off"` is unnecessary.

Reply:
No specific concerns, this can be left as is. Thanks for the comments.

Reply:
so I need to remove `strategy == "off"`? cc @dndx

Reply:
I talked with dndx, I think we should remove `strategy == "off"`, which will make code more clear.

Reply:
I had second thought and it still seems that hashing the `cache_key` makes little sense, and this could make debugging shdict related caching issue harder. Is there a reason why we do **not** do this in place where we actually set keys inside LMDB?

The argument "Wouldn't keeping the behavior consistent across all code paths be better here?" would not apply, because I feel the `cache_key` function is simply the wrong place to do it.

Reply:
No, we need to query LMDB by key, only changing set is not enough.